### DISCUSSION TITLE
Documentation troubles

### DISCUSSION THREAD

**Discussion by Gouvernathor** [TwitchIO/TwitchIO#402] [Questions]
Some issues I had while reading the documentation.

- the twitchio.ext.commands.command and twitchio.ext.commands.cooldown decorators (notice the lowercase) are missing. The capitalized versions, the classes, are present, and then an example shows the decorators. The decorators do take signatures, and the classes too, but while the signature of the cooldown class and decorator are identical, the command class takes parameters and the command decorator does not.
- Are there interferences with the ordering and/or cumulation of decorators ? The example shows the cooldown being applied before command, but is that mandatory (I think there are such interferences in some stdlib decorators, so I think it's worth mentioning) ? Is it possible to add several cooldowns of different values on the same command ?
- the command decorator prevents the decorated functions from taking extraneous optional parameters, i.e to call a command's method with an optional parameter tweaking its behavior, from another command's method. Documenting that would help.
- about the cooldown decorators, how do the `subscriber` and `mod` cooldowns work ? Do they restrict the use of the command by the mods only, and then limit their use of it ? Or do they rather let peones use it at will and only limit the mods/subscribers ?

If I find others I can add them here.

[a comment by an automated account is not shown]

**Comment by IAmTomahawkx** [comment]
This doesn't seem to be an issue with the library. I'll convert this to a discussion.

**Reply by Gouvernathor** [reply]
It's an issue with the docs, and since the docs are part of the repo I think that makes it an issue with the library. Most repos do it that way as far as I know (cpython for instance), but to each their own I guess.
